Tools/autotest/rover.py

```diff
@@ -442,15 +444,17 @@ def do_get_banner(self):
         raise MsgRcvTimeoutException("banner not received")
 
     def drive_brake_get_stopping_distance(self, speed):
-        # measure our stopping distance:
-        old_cruise_speed = self.get_parameter('CRUISE_SPEED')
-        old_accel_max = self.get_parameter('ATC_ACCEL_MAX')
+        '''measure our stopping distance'''
+
+        self.context_push()
 
         # controller tends not to meet cruise speed (max of ~14 when 15
         # set), thus *1.2
-        self.set_parameter('CRUISE_SPEED', speed*1.2)
         # at time of writing, the vehicle is only capable of 10m/s/s accel
-        self.set_parameter('ATC_ACCEL_MAX', 15)
+        self.set_parameters({
+            'CRUISE_SPEED': speed*1.2,
```

Comment:
this should probably be WP_SPEED.  CRUISE_SPEED use to control the vehicle's speed in Auto mode but now we use WP_SPEED.

Reply:
These tests are done in `STEERING` - doesn't that make `CRUISE_SPEED` the relevant parameter?

Reply:
@peterbarker, ah, ok.  So CRUISE_SPEED is used in all manual modes as a way to calculate the vehicle's top speed.  So, if the intention is to increase the vehicle's top speed during the test (without changing the throttle input by the virtual human driver) this is one way to do that.  Another way though is just to raise the throttle.

Reply:
> @peterbarker, ah, ok. So CRUISE_SPEED is used in all manual modes as a way to calculate the vehicle's top speed. So, if the intention is to increase the vehicle's top speed during the test (without changing the throttle input by the virtual human driver) this is one way to do that. Another way though is just to raise the throttle.

Looks like this function is, "give me a distance in metres, stopping distance given an initial speed".  Achieiving that speed is done by setting the cruise speed parameter and full-throttle.  I suspect a better way of doing this with modern Rover would be to use guided mode sending in velocity commands....

Still, really just cleaning things up (and making things marginally faster; setting parameters as a batch is a little faster than doing them individually).  These tests were written before "contexts" existed in the suite, so they were reverting the vehicle state themselves in the old, clunky, manner.

Reply:
yup, if anything it's an existing issue so i'm ok with it. txs